pkg/manifestreader/path.go

```diff
@@ -0,0 +1,53 @@
+// Copyright 2020 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package manifestreader
+
+import (
+	"k8s.io/cli-runtime/pkg/resource"
+)
+
+// PathManifestReader reads manifests from the provided path
+// and returns them as Info objects. The returned Infos will not have
+// client or mapping set.
+type PathManifestReader struct {
+	Path string
+
+	ReaderOptions
+}
+
+// Read reads the manifests and returns them as Info objects.
+func (p *PathManifestReader) Read() ([]*resource.Info, error) {
+	validator, err := p.Factory.Validator(p.Validate)
+	if err != nil {
+		return nil, err
+	}
+
+	fileNameOptions := &resource.FilenameOptions{
+		Filenames: []string{p.Path},
+		Recursive: true,
+	}
+
+	result := p.Factory.NewBuilder().
+		Local().
+		Unstructured().
+		Schema(validator).
+		ContinueOnError().
+		FilenameParam(false, fileNameOptions).
```

Comment:
Should we name this `false` parameter as a variable first to document it; othewise, we can just add a comment.

Reply:
Done